### DISCUSSION TITLE
Login fails with IMAGE_TO_RAM

### DISCUSSION THREAD

**Discussion by stevehol** [ltsp/ltsp#269] [Q&A]
I'm trying to improve app startup times which can be as long as 18 secs for Libre Office Writer and Firefox.
I've got everything in a single 60GB partition on the server and create a chrootless client image.
My test client has 8GB RAM and a 240GB SSD as swap. Logins are OK without IMAGE_TO_RAM.
Then I put IMAGE_TO_RAM=1 under [clients] in ltsp.conf, and did ltsp initrd.
It takes just under 2 minutes to boot, which is OK, but logins all fail with ".Login incorrect." after 60 seconds. Logins fail the same without the SSD.
Is there something else I need to do?

**Comment by alkisg** [comment]
For image caching, you can just run a dd command in the background while the client boots, like `dd if=/run/initramfs/ltsp/... of=/dev/null bs=1M ... &`, we discussed this earlier in some issue or pull request. It's usually better for this use case, as it allows the client to reclaim the RAM in low memory situations.

For the login failure, wait for feedback from the IMAGE_TO_RAM developer, @kvaps; maybe you also need to set the server as a parameter.

**Comment by kvaps** [comment]
Hi, I'm not using upstream LTSP project and ssh authentication as well, so I can't say for sure.

There is also two different implementation for upstream and `ltsp-cloud` projects, more details in this PR https://github.com/ltsp/ltsp/pull/103

I can advice you try install `ltsp-cloud` package from [proposed PPA](https://ltsp.org/advanced/proposed-ppa/), regenerate image and test authentication on it, it is little outdated from upstream but my features are well tested on it https://ltsp.org/advanced/proposed-ppa/

**Comment by stevehol** [comment]
Hi thanks for your help.
@alkisg - you were right, I just needed to give the SERVER IP in the [clients] section of lts.conf and logins are fixed.
BTW after using LTSP for many years I'm very impressed with LTSP20 - you've done a great job, thank you!
@ kvaps - thanks for you advice, I'll take a look later.
